app/assets/javascripts/restricted_notes/restricted_notes_page_container.js

```diff
@@ -0,0 +1,129 @@
+    displayName: 'RestrictedNotesPageContainer',
+
+    propTypes: {
+      nowMomentFn: React.PropTypes.func.isRequired,
+      serializedData: React.PropTypes.object.isRequired,
+
+      // for testing
+      actions: React.PropTypes.shape({
+        onClickSaveNotes: React.PropTypes.func
+      }),
+      api: PropTypes.api
+    },
+
+    componentWillMount: function(props, state) {
+      this.api = this.props.api || new Api();
+    },
+
+    getInitialState: function() {
+      var serializedData = this.props.serializedData;
+
+      return {
+        // context
+        currentEducator: serializedData.currentEducator,
+        // constants
+        educatorsIndex: serializedData.educatorsIndex,
+        eventNoteTypesIndex: serializedData.eventNoteTypesIndex,
+        // data
+        feed: serializedData.feed,
+        student: serializedData.student,
+        // ui
+        // This map holds the state of network requests for various actions.  This allows UI components to branch on this
+        // and show waiting messages or error messages.
+        // The state of a network request is described with null (no requests in-flight),
+        // 'pending' (a request is currently in-flight),
+        // and 'error' or another value if the request failed.
+        // The keys within `request` hold either a single value describing the state of the request, or a map that describes the
+        // state of requests related to a particular object.
+        // For example, `saveService` holds the state of that request, but `saveNotes` is a map that can track multiple active
+        // requests, using `serviceId` as a key.
+        requests: {
+          saveNotes: {},
+          saveService: null,
+          discontinueService: {}
+        }
+      };
+    },
+
+    onClickSaveNotes: function(eventNoteParams) {
+      // All notes taken on this page should be restricted.
+      var eventNoteParams = merge(eventNoteParams, {is_restricted: true});
+
+      var saveNotes = {}
+      saveNotes[eventNoteParams.id] = 'pending';
+      this.setState({ requests: merge(this.state.requests, { saveNotes: saveNotes}) });
+      this.api.saveNotes(this.state.student.id, eventNoteParams)
+        .done(this.onSaveNotesDone)
+        .fail(this.onSaveNotesFail);
+    },
+
+    onSaveNotesDone: function(response) {
+      var updatedEventNotes = this.state.feed.event_notes.concat([response]);
+      var updatedFeed = merge(this.state.feed, { event_notes: updatedEventNotes });
+      this.setState({
+        feed: updatedFeed,
+        requests: merge(this.state.requests, { saveNotes: {} })
+      });
+    },
+
+    onSaveNotesFail: function(request, status, message) {
+      var saveNotes = {}
+      saveNotes[eventNoteParams.id] = 'pending';
+      this.setState({
+        requests: merge(this.state.requests, {saveNotes: saveNotes})
+      });
+    },
+
+    getNotesHelpContent: function(){
```

Comment:
I just noticed this - if we need HTML from JavaScript we should be using React to create it rather than writing it in strings.  As the comment notes, this creates opportunities for XSS that we can avoid altogether.  I'll open a separate issue to refactor this to React elements and update `HelpBubble`, since that seems like a great starter issue for someone! :)